From 790c94e3240549de61881aebb94d4d933252e540 Mon Sep 17 00:00:00 2001
From: Danilo de Paula <ddepaula@redhat.com>
Date: Mon, 20 May 2019 18:29:57 +0100
Subject: [PATCH] target/i386: define md-clear bit

RH-Author: Danilo de Paula <ddepaula@redhat.com>
Message-id: <20190520182957.26425-1-ddepaula@redhat.com>
Patchwork-id: 88110
O-Subject: [RHEL-8 + RHEL-AV qemu-kvm PATCH] target/i386: define md-clear bit
Bugzilla: 1703297 1703304 1703310 1707274
RH-Acked-by: Daniel P. Berrange <berrange@redhat.com>
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>
RH-Acked-by: John Snow <jsnow@redhat.com>

From: Paolo Bonzini <pbonzini@redhat.com>

BZ: 1703310
BZ: 1703304
BZ: 1703297
BZ: 1707274
branch: rhel-av-8.1.0/master-4.0.0

BZ: 1705851
BZ: 1704542
BZ: 1704538
BZ: 1704534
branch: rhel-av-8.0.1

BZ: 1703308
BZ: 1703302
branch: rhel-8.1.0

md-clear is a new CPUID bit which is set when microcode provides the
mechanism to invoke a flush of various exploitable CPU buffers by invoking
the VERW instruction.

Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
Signed-off-by: Danilo C. L. de Paula <ddepaula@redhat.com>
---
 target/i386/cpu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/target/i386/cpu.c b/target/i386/cpu.c
index 6472cd2..3886464 100644
--- a/target/i386/cpu.c
+++ b/target/i386/cpu.c
@@ -1076,7 +1076,7 @@ static FeatureWordInfo feature_word_info[FEATURE_WORDS] = {
         .feat_names = {
             NULL, NULL, "avx512-4vnniw", "avx512-4fmaps",
             NULL, NULL, NULL, NULL,
-            NULL, NULL, NULL, NULL,
+            NULL, NULL, "md-clear", NULL,
             NULL, NULL, NULL, NULL,
             NULL, NULL, NULL, NULL,
             NULL, NULL, NULL, NULL,
-- 
1.8.3.1

